pkg/client/typed/dynamic/client.go

```diff
@@ -79,28 +84,54 @@ func NewClient(conf *restclient.Config) (*Client, error) {
 	return &Client{cl: cl}, nil
 }
 
-// Resource returns an API interface to the specified resource for
-// this client's group and version. If resource is not a namespaced
-// resource, then namespace is ignored.
+// Resource returns an API interface to the specified resource for this client's
+// group and version. If resource is not a namespaced resource, then namespace
+// is ignored.
 func (c *Client) Resource(resource *unversioned.APIResource, namespace string) *ResourceClient {
-	return &ResourceClient{
+	rc := &ResourceClient{
 		cl:       c.cl,
 		resource: resource,
 		ns:       namespace,
 	}
+	return rc
+}
+
+// ParameterCodec wraps a parameterCodec around the Client.
+func (c *Client) ParameterCodec(parameterCodec runtime.ParameterCodec) *ClientWithParameterCodec {
```

Comment:
Make a copy sounds heavy, sort of defeated effort of the client pool.

What's wrong with this wrapper approach?